### PR TITLE
HOLO-1419: Add minting NFT flow

### PR DESCRIPTION
**Pull request by feliperodriguess**
## Describe Changes

Changes made to the package: **sdk**.

I made it better by doing ...

- Create `CxipERC721` contract class
- Create `HolographDropERC721` contract class
- Add `mint`, `moeMint`, `_estimateGasForMintingNft`, `_estimateGasForMintingMoeNft` methods to the NFT asset class
- Refactor function gas estimation call
- Scaffold `CxipERC721` &  `HolographDropERC721` contract tests
- Add general improvements
- Add tests to the `NFT` asset class, mostly on the minting flow
- Add tests to the `MoeNFT` asset class

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All tests are passing

### CODE REVIEW

File: packages/sdk/src/assets/collection-moe.ts [hunk 1]
```diff
@@ -57,10 +47,10 @@ export class HolographMoeERC721DropV1 {
...
     {collectionInfo, nftInfo, primaryChainId, saleConfig}: CreateHolographMoe,
   ) {
-    this.collectionInfo = collectionInfoSchema.parse(collectionInfo)
-    this.nftInfo = nftInfoSchema.parse(nftInfo)
-    this.saleConfig = holographMoeSaleConfigSchema.parse(saleConfig)
-    this.primaryChainId = primaryChainIdSchema.parse(primaryChainId)
+    this.collectionInfo = validate.collectionInfo.parse(collectionInfo)
+    this.nftInfo = validate.nftInfo.parse(nftInfo)
+    this.saleConfig = validate.saleConfig.parse(saleConfig)
+    this.primaryChainId = validate.primaryChainId.parse(primaryChainId)
```
Comment (sogoiii) [inline packages/sdk/src/assets/collection-moe.ts:53]: I like the way this reads

File: packages/sdk/src/assets/nft.ts [hunk 2]
```diff
@@ -1,21 +1,37 @@
...
-    this.metadata = metadata
+  private cxipERC721: CxipERC721
+  private holographDropERC721: HolographDropERC721
+
+  constructor(configObject: HolographConfig, {chainId, collectionAddress, metadata}: CreateNft) {
+    this.chainId = validate.chainId.parse(chainId)
+    this.collectionAddress = validate.collectionAddress.parse(collectionAddress) as Address
+    this.metadata = validate.metadata.parse(metadata)
+    const config = Config.getInstance(configObject)
+    const cxipERC721 = new CxipERC721(config, this.collectionAddress)
```
Comment (sogoiii) [inline packages/sdk/src/assets/nft.ts:31]: I do not like that the nft has both kinds of collections defined within it. I can see that a dev can confuse if this nft suppose to be minted to a legacy collection or an MOE collection.

File: packages/sdk/src/contracts/holograph-base.contract.ts [hunk 3]
```diff
@@ -143,6 +149,19 @@ export class HolographBaseContract {
 
   /**
    * @readonly
+<<<<<<< HEAD
```
Comment (sogoiii) [inline packages/sdk/src/contracts/holograph-base.contract.ts:152]: This diff comment should be removed.

Comment (feliperodriguess) [reply packages/sdk/src/contracts/holograph-base.contract.ts]: Done

File: packages/sdk/src/assets/nft.ts (hunk 2 above)
Comment (feliperodriguess) [reply packages/sdk/src/assets/nft.ts:31]: but we have two different methods `mint` and `moeMint`, do you want to create a separate class for the moe nft?

Comment (feliperodriguess) [reply packages/sdk/src/assets/nft.ts:33]: Done

**Review by sogoiii — APPROVED** [review]
👍

File: packages/sdk/src/constants/gas-controllers/moe-nft-mint.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    gasLimitMultiplier: 110,
+    gasPriceMultiplier: 110,
+  },
+  [networks.localhost2.chain]: {
+    gasPrice: undefined,
+    gasLimit: undefined,
+    gasLimitMultiplier: 110,
+    gasPriceMultiplier: 110,
+  },
+}
```
Comment (sogoiii) [inline packages/sdk/src/constants/gas-controllers/moe-nft-mint.ts:154]: add `as const satisfies Record<SupportedChainIds, GasParams>`  like i did in this pr (https://github.com/holographxyz/holograph-app/pull/937/files)

**Review by sogoiii — CHANGES_REQUESTED** [review]

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**